subprojects/core/src/main/java/org/gradle/initialization/DefaultProjectDescriptor.java

```diff
@@ -105,6 +112,16 @@ void setPath(Path path) {
     }
 
     public String getBuildFileName() {
+        if (buildFileName == null) {
```

Comment:
:x: I would change the code to extract the file name from the build file instead. See 0d0e83d837ef877e8205cb9200ceda4aa2e66962 for the rationale.